### PR TITLE
HOSTEDCP-1373: ARO HCP - Add capability for Azure VMs to be created with ephemeral disks

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
Adds the capability to enable ephemeral disks for the OS disks in the Azure VMs. In addition, the ability to set the disk storage account type for the OS disks through the CLI was added in this PR as part of the prerequisite work.

**Which issue(s) this PR fixes**:
[HOSTEDCP-1373](https://issues.redhat.com/browse/HOSTEDCP-1373)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [x] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by bryan-cox** [comment]
@csrwng @sjenning - I made some changes to the default OS disk size for azure nodes [here](https://github.com/openshift/hypershift/blob/2625cab6aebd810c654b3eb9210a45d1f3582c95/api/hypershift/v1beta1/nodepool_types.go#L861). I saw there was some discussion about this previously with Alvaro here [on this PR](https://github.com/openshift/hypershift/pull/1097#issuecomment-1059274687).

The default I set it to is what CAPZ defaults to. The previous default value is too large for ephemeral disks.

File: docs/content/how-to/azure/create-azure-cluster-with-options.md [hunk 1]
```diff
@@ -100,3 +100,38 @@ spec:
...
   replicas: 0
 ```
+
+## Enabling Ephemeral OS Disks on Azure VMs
+To enable the ephemeral OS disk option on the Azure VMs in your HostedCluster, set the `enable-ephemeral-disk` flag to true.
+
+!!! important
+
+    Ephermeral OS disks are not available in every region or for every instance type. You may also need to adjust 
+    the disk size accordingly; to adjust the disk size use the `disk-storage-account-type` flag as shown in the example below.
```
Comment (csrwng) [inline docs/content/how-to/azure/create-azure-cluster-with-options.md:110]: Just curious, how does the disk storage account type adjust the disk size?

**Comment by csrwng** [comment]
> The default is what CAPZ defaults to. The previous default value is too large for ephemeral disks.

Could we set the default (at least in the CLI) based on whether you are using ephemeral disks?

File: docs/content/how-to/azure/create-azure-cluster-with-options.md (hunk 1 above)
Comment (bryan-cox) [reply docs/content/how-to/azure/create-azure-cluster-with-options.md:110]: It doesn't. I typo-ed that and meant disk storage account type.

**Comment by bryan-cox** [comment]
Retesting RHTAP
/retest

**Pull request merged**